cleanlab/multiannotator.py

```diff
@@ -697,70 +727,88 @@ def get_active_learning_scores_ensemble(
+        if labels_multiannotator.apply(lambda s: len(s.dropna()) == 1, axis=1).all():
+            # do not temp scale for single annotator case, temperature is defined here for later use
+            optimal_temp = np.full(len(pred_probs), 1.0)
 
-        avg_pred_probs = np.mean(pred_probs, axis=0)
-        consensus_label = get_majority_vote_label(
-            labels_multiannotator=labels_multiannotator,
-            pred_probs=avg_pred_probs,
-            verbose=False,
-        )
-        quality_of_consensus_labeled = get_label_quality_scores(consensus_label, avg_pred_probs)
-        model_weight = np.full(len(pred_probs), 1)
-        annotator_weight = np.full(labels_multiannotator.shape[1], 1)
-        avg_annotator_weight = np.mean(annotator_weight)
+            assert_valid_inputs_multiannotator(
+                labels_multiannotator, pred_probs, ensemble=True, allow_single_label=True
+            )
 
-    else:
-        optimal_temp = np.full(len(pred_probs), np.NaN)
-        for i in range(len(pred_probs)):
-            curr_pred_probs = pred_probs[i]
-            curr_optimal_temp = find_best_temp_scaler(labels_multiannotator, curr_pred_probs)
-            pred_probs[i] = temp_scale_pred_probs(curr_pred_probs, curr_optimal_temp)
-            optimal_temp[i] = curr_optimal_temp
-
-        multiannotator_info = get_label_quality_multiannotator_ensemble(
-            labels_multiannotator,
-            pred_probs,
-            return_annotator_stats=False,
-            return_detailed_quality=False,
-            return_weights=True,
-        )
+            avg_pred_probs = np.mean(pred_probs, axis=0)
+            consensus_label = get_majority_vote_label(
+                labels_multiannotator=labels_multiannotator,
+                pred_probs=avg_pred_probs,
+                verbose=False,
+            )
+            quality_of_consensus_labeled = get_label_quality_scores(consensus_label, avg_pred_probs)
+            model_weight = np.full(len(pred_probs), 1)
+            annotator_weight = np.full(labels_multiannotator.shape[1], 1)
+            avg_annotator_weight = np.mean(annotator_weight)
 
-        quality_of_consensus_labeled = multiannotator_info["label_quality"][
-            "consensus_quality_score"
-        ]
-        model_weight = multiannotator_info["model_weight"]
-        annotator_weight = multiannotator_info["annotator_weight"]
-        avg_annotator_weight = np.mean(annotator_weight)
-
-    # compute scores for labeled data
-    active_learning_scores = np.full(len(labels_multiannotator), np.nan)
-    for i in range(len(active_learning_scores)):
-        annotator_labels = labels_multiannotator.iloc[i]
-        active_learning_scores[i] = np.average(
-            (quality_of_consensus_labeled[i], 1 / num_classes),
-            weights=(
-                np.sum(annotator_weight[annotator_labels.notna()]) + np.sum(model_weight),
-                avg_annotator_weight,
-            ),
+        else:
```

Comment:
[nit] for such if/else branches where there is a lot of code in between the if/elif/else statements, I like to add comment to remind what situation we are in for each one. I.e. for this one, would add comment:

# no labeled data provided so do not estimate temperature and model/annotator weights